src/oasis-nexus/api.ts

```diff
@@ -433,6 +443,64 @@ export const useGetRuntimeAccountsAddress: typeof generated.useGetRuntimeAccount
   } as any
 }
 
+const MAX_LOADED_ADDRESSES = 100
+const MASS_LOAD_INDEXES = [...Array(MAX_LOADED_ADDRESSES).keys()]
+
+type RuntimeTarget = {
+  network: Network
+  layer: Runtime
+  address: string
+}
+
+export const useGetRuntimeAccountsAddresses = (
+  targets: RuntimeTarget[] | undefined = [],
+  queryOptions: { enabled: boolean },
+) => {
+  const queries = MASS_LOAD_INDEXES.map((i): RuntimeTarget | undefined => targets[i]).map(target =>
+    // The number of iterations is constant here, so we will always call the hook the same number.
+    // eslint-disable-next-line react-hooks/rules-of-hooks
+    useGetRuntimeAccountsAddress(
+      target?.network ?? Network.mainnet,
+      target?.layer ?? Layer.emerald,
+      target?.address ?? '',
+      {
+        query: { enabled: queryOptions.enabled && !!target?.address },
+      },
+    ),
+  )
+  return {
+    isLoading: !!targets?.length && queries.some(query => query.isLoading && query.fetchStatus !== 'idle'),
+    isError: queries.some(query => query.isError),
+    isFetched: !queries.some(query => !query.isFetched),
```

Comment:
This doesn't ignore disabled MASS_LOAD queries. Is it always false?

Reply:
We don't actually use or need the values calculated here. Removed.